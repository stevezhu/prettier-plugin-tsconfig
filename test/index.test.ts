import { readFile } from 'fs/promises';
import { resolve } from 'path';
import { format } from 'prettier';
import { expect, test } from 'vitest';

import * as plugin from '../src/index.js';

test.each([
  ['sorts tsconfig', 'tsconfig.test.json'],
  [
    'sorts unknown fields at end in lexical order',
    'tsconfig.unknownFields.json',
  ],
])('%s', async (_, filename) => {
  // Fixtures are prefixed with an '_' so that they don't get formatted by this plugin
  const filepath = resolve(__dirname, 'fixtures', `_${filename}`);
  const tsconfigTest = await readFile(filepath, 'utf8');
  await expect(
    // set the filepath without the '_' prefix here so that the plugin knows to format the file
    format(tsconfigTest, { filepath: `./${filename}`, plugins: [plugin] }),
  ).resolves.toMatchSnapshot(filename);
});
