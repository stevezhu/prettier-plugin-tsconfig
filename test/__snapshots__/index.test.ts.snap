// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sorts tsconfig > tsconfig.test.json 1`] = `
"{
  \\"$schema\\": \\"https://json.schemastore.org/tsconfig\\",
  \\"compilerOptions\\": {
    \\"exactOptionalPropertyTypes\\": true,
    \\"noImplicitOverride\\": true,
    \\"noImplicitReturns\\": true,
    \\"noPropertyAccessFromIndexSignature\\": true,
    \\"noUncheckedIndexedAccess\\": true,
    \\"noUnusedLocals\\": true,
    \\"strict\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"esModuleInterop\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"isolatedModules\\": true,
    \\"target\\": \\"ESNext\\",
    \\"skipLibCheck\\": true
  }
}
"
`;

exports[`sorts unknown fields at end in lexical order > tsconfig.unknownFields.json 1`] = `
"{
  \\"$schema\\": \\"https://json.schemastore.org/tsconfig\\",
  \\"exclude\\": [\\"mockfile.ts\\"],
  \\"compilerOptions\\": {
    \\"strict\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"isolatedModules\\": true,
    \\"target\\": \\"ESNext\\",
    \\"unknownField\\": 1,
    \\"unknownField2\\": 2,
    \\"unknownField3\\": { \\"unknownNestedField1\\": 1, \\"unknownNestedField2\\": 2 },
    \\"unknownField4\\": [3, 2, 1]
  },
  \\"display\\": \\"Unknown Fields Test\\"
}
"
`;
